presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -613,7 +613,6 @@ public void rollback()
     {
```

Comment:
We can also handle the case when `icebergTable.specs().values().stream().allMatch(spec -> spec.fields().containsAll(fields))` is false. Currently,
```
        if (newDomain.equals(table.getPredicate().intersect(table.getEnforcedPredicate()))) {
            return Optional.empty();
        }
```
Even in this case the pushed predicate can be used in table scans.

Reply:
Good point. Thinking more on this, I think there might be a better approach to this. I've updated the PR with this new approach.